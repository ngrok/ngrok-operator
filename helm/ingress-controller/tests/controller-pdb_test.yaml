suite: test controller-pdb
templates:
- controller-pdb.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
- it: should match snapshot
  set:
    podDisruptionBudget:
      create: true
  asserts:
  - matchSnapshot: {}
- it: Defaults to 1 maxUnavailable
  template: controller-pdb.yaml
  documentIndex: 0 # Document 0 is the deployment since its the first template
  set:
    podDisruptionBudget:
      create: true
  asserts:
  - equal:
      path: spec.maxUnavailable
      value: 1
- it: Allows changing the minAvailable value
  template: controller-pdb.yaml
  documentIndex: 0 # Document 0 is the deployment since its the first template
  set:
    podDisruptionBudget:
      create: true
      minAvailable: 1
      maxUnavailable: ""
  asserts:
  - equal:
      path: spec.minAvailable
      value: 1
