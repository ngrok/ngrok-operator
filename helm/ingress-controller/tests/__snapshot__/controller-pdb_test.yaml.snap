should match snapshot:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-ingress-controller
        app.kubernetes.io/part-of: kubernetes-ingress-controller
        app.kubernetes.io/version: 0.10.3
        helm.sh/chart: kubernetes-ingress-controller-0.12.3
      name: test-release-kubernetes-ingress-controller-controller-pdb
      namespace: test-namespace
    spec:
      maxUnavailable: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: controller
          app.kubernetes.io/instance: test-release
          app.kubernetes.io/name: kubernetes-ingress-controller
